### ISSUE TITLE
HAVE_AUFS not defined

### ISSUE THREAD

**Issue by bkauler** [ghedo/pflask#25]
Hi,
I have a Linux kernel (4.14.71) with aufs. Overlayfs is disabled.

The pflask docs state, for the "--mount" option:

> Stacks the host root_dir directory on top of the container’s dest directory using either AuFS or OverlayFS depending on what is found at compile-time.

File mount.c tests "HAVE_AUFS", however, this variable is not set anywhere. I am not familiar with Waf, looked at build/c4che/_cache.py, has a lot of variables but not HAVE_AUFS.

Do I have to explicitly set HAVE_AUFS? If so, is there a particular way to do it for waf?

Regards,
Barry

**Comment by bkauler** [comment]
Hmmm, I looked back through the commits, and found that the guts of supporting aufs has been removed, even though the test for "HAVE_AUFS" is still in mount.c.

Puppy Linux and derivatives still use aufs, and that is not likely to change. I maintain Quirky Linux and EasyOS, forks of Puppy. I experimented with overlayfs about a year ago, and found a serious fault, that was not going to get fixed anytime soon -- can't recall what it was. Also, aufs supports direct writing to lower layers, which Puppy requires.

I am using pflask with the --chroot option, so it is not an issue. However, might have a look at reverting that removal of aufs, and fix it.

Just a note on the side: Puppy Linux and derivatives do not use systemd, they use the simple Busybox init.

**Comment by ghedo** [comment]
Yeah, I removed AUFS support a while ago because it was dropped from the Debian kernels and I had no way to test whether the code was still working or not (so it was basically unmaintained).
